front/pages/api/w/[wId]/index.ts

```diff
@@ -83,16 +85,27 @@ async function handler(
         });
       }
 
-      if (req.body.name) {
+      if ("name" in body) {
         await w.update({
-          name: req.body.name,
+          name: body.name,
         });
-        owner.name = req.body.name as string;
+        owner.name = body.name;
       } else {
-        await w.update({
-          allowedDomain: req.body.allowedDomain,
-        });
-        owner.allowedDomain = req.body.allowedDomain as string | null;
+        const user = auth.user();
```

Comment:
This route is expecting a user so you can extract it after line 38 and share the error `if (!user || !owner)`